### PR TITLE
Fixes for 1.0

### PR DESCRIPTION
**Pull request by quinnj**
This fixes building Conda.jl on 1.0.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ const ROOTENV = "$(escape_string(ROOTENV))"
 const MINICONDA_VERSION = "$(escape_string(MINICONDA_VERSION))"
 """
 
-if !isfile("deps.jl") || readstring("deps.jl") != deps
+if !isfile("deps.jl") || String(read("deps.jl")) != deps
```
Comment (fredrikekre) [inline deps/build.jl]: `read("deps.jl", String)`?

**Comment by jekbradbury** [comment]
`@isdefined` isn't defined on 0.6, while both REQUIRE and Travis indicate that Conda master is meant to continue working on 0.6, so those lines may need to be guarded with `if VERSION`

**Comment by devgeist42** [comment]
How about
```
module DefaultDeps
    if VERSION < v"0.7"
        macro isdefined(x)
            isdefined(x)
        end
    end
    if isfile("deps.jl")
        include("deps.jl")
    end
    if !(@isdefined ROOTENV)
        const ROOTENV = abspath(dirname(@__FILE__), "usr")
    end
    if !(@isdefined MINICONDA_VERSION)
        const MINICONDA_VERSION = "3"
    end
end
```

**Comment by jekbradbury** [comment]
I went with something similar; will merge if Travis is happy because no IPython on 1.0 is 😢.

File: deps/build.jl [hunk 2]
```diff
@@ -4,10 +4,10 @@ module DefaultDeps
     if isfile("deps.jl")
         include("deps.jl")
     end
-    if !isdefined(:ROOTENV)
+    if !isdefined(@__MODULE__, :ROOTENV)
```
Comment (yuyichao) [inline deps/build.jl]: Errr, I think you need Compat... The using above is not in the right module iiuc.

**Comment by jekbradbury** [comment]
The remaining issue is that the Appveyor script uses `julia -e 'versioninfo()`, which needs `using InteractiveUtils` on 1.0. Should Conda switch to the Appveyor.jl version of the script?

**Comment by stevengj** [comment]
Merging this as-is, so that people can use Conda.jl under 1.0.   Fixing AppVeyor can be a separate PR.

**Comment by stevengj** [comment]
Merging this as-is, so that people can use Conda.jl under 1.0.   Fixing AppVeyor can be a separate PR.

**Comment by fredrikekre** [comment]
Can we make a new release such that people can use IJulia on v1.0.

**Pull request merged**